File: src/exercises/clock/go/template.go
package main

type Clock struct {
	totalMinutes int
}

func NewClock(hour, minute int) Clock {
	// Your code here
	return Clock{}
}

func (c Clock) String() string {
	// Your code here
	return "00:00"
}

func (c Clock) Add(minutes int) Clock {
	// Your code here
	return c
}

func (c Clock) Subtract(minutes int) Clock {
	// Your code here
	return c
}

File: src/exercises/clock/ocaml/template.ml
type clock = { total_minutes : int }

let create (hour : int) (minute : int) : clock =
  (* Your code here *)
  { total_minutes = 0 }

let add (minutes : int) (c : clock) : clock =
  (* Your code here *)
  c

let sub (minutes : int) (c : clock) : clock =
  (* Your code here *)
  c

let to_string (c : clock) : string =
  (* Your code here *)
  "00:00"
